ccm_web/server/src/invitation/cirrus-invitation.service.ts

```diff
@@ -0,0 +1,70 @@
+import FormData from 'form-data'
+import { randomUUID } from 'crypto'
+import { HttpService } from '@nestjs/axios'
+import { ConfigService } from '@nestjs/config'
+import { Injectable } from '@nestjs/common'
+import baseLogger from '../logger'
+import { CanvasUserLoginEmail } from '../canvas/canvas.interfaces'
+import { Config, InvitationConfig } from '../config'
+import { InvitationAPIError } from './invitation.errors'
+import { lastValueFrom } from 'rxjs'
+import { CirrusInvitationResponse } from './cirrus-invitation.interfaces'
+import axios from 'axios'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+@Injectable()
+export class CirrusInvitationService {
+  url: string
+  entityID: string
+  sponsorName: string
+  key: string
+  secret: string
+
+  constructor (
+    private readonly configService: ConfigService<Config, true>,
+    private readonly httpService: HttpService
+  ) {
+    const invitationConfig = configService.get('invitation', { infer: true }) as InvitationConfig
+    this.url = invitationConfig.apiURL
```

Comment:
I have some comment on how this URL need to change here https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/247#discussion_r809084249

Reply:
🏁